### PR TITLE
Make loader platform specific

### PR DESCRIPTION
**Pull request by grabbou**
Fixes an issue when running with `--platform all` makes assets overwrite each other (Android overwrites iOS and then, iOS keeps returning Android back icon for instance).

### CODE REVIEW

**Comment by grabbou** [comment]
This is not yet ready for merging as I am working on unit tests for that file & adding few more comments here and there.

Feedback appreciated.

**Comment by satya164** [comment]
I think the simplest fix is to change `path.join('assets', url, name);` to `path.join('assets', query.bundle ? '' : query.platform, url, name);` and `path.join('/assets', url)` to `path.join('assets', query.bundle ? '' : query.platform, url))` (store the later in a variable and reuse)

This change changes the meaning of the `url` variable which was supposed to be the relative URL to the destination. It will likely break bundling assets for iOS. Not sure what else it will break since it was the original assumption while writing the code. 

Can we do the change I mentioned above instead? I'll add some comments to the code so it's easier to understand.

**Comment by grabbou** [comment]
> This change changes the meaning of the url variable which was supposed to be the relative URL to the destination. It will likely break bundling assets for iOS. Not sure what else it will break since it was the original assumption while writing the code.

Not really sure, `url` is used only in two places, in those path.join calls. How concept of `relative url` fits in there?

**Comment by satya164** [comment]
So it won't break anything else, but it'll definitely break bundling assets for iOS. Can we do the change I mentioned instead? I'll add comments later.

**Comment by grabbou** [comment]
> So it won't break anything else, but it'll definitely break bundling assets for iOS

It shouldn't break bundling for any platform since we don't change the code when `query.bundle` is true. Your code suggestion is just moving where `path.join` happens.

File: src/loaders/assetLoader.js [hunk 1]
```diff
@@ -111,18 +115,20 @@ module.exports = async function assetLoader() {
                   throw new Error(`Unknown scale ${scale} for ${filepath}`);
               }
 
+              const longname = `${`${url.replace(/\//g, '_')}_${filename}`
```
Comment (satya164) [inline src/loaders/assetLoader.js]: Why do this multiple times instead of once?

Comment (grabbou) [inline src/loaders/assetLoader.js]: Just wanted to scope it for now. Ultimately I want it to go to a separate function, so that will be gone.

**Comment by satya164** [comment]
This will break iOS because iOS expects the assets to be folder inside `assets` which this PR removes.

File: src/loaders/assetLoader.js (hunk 1 above)
Comment (satya164) [inline src/loaders/assetLoader.js]: Why does it need scoping? It's the same name for every file.

**Comment by grabbou** [comment]
> This will break iOS because iOS expects the assets to be folder inside assets which this PR removes.

While bundling, Xcode specifies `--assetsDest` that already points to that folder.

File: src/loaders/assetLoader.js (hunk 1 above)
Comment (grabbou) [inline src/loaders/assetLoader.js]: It doesn't. It's WIP and that was moved down to help me understand usage. We can ignore this line.

**Comment by grabbou** [comment]
> This will break iOS because iOS expects the assets to be inside a folder assets which this PR removes.

@satya164 Yeah, I'll revert that change. Sorry for that. I had a folder `assets` already and it got me confused.

**Comment by satya164** [comment]
Also I think `assets/ios/path/to/my/images/image.png` will be better rather than `path/to/my/images/ios/image.png`, because the former preserves the actual path to the file and prepends a path, as opposed to modifying the path.

**Comment by satya164** [comment]
`url = path.join('assets', query.bundle ? '' : query.platform, path.relative(...))` seems much better

**Comment by satya164** [comment]
Oh, `url` is also used in `longname`, so `url` needs to be the relative path.

**Comment by grabbou** [comment]
I am going to resubmit it again in a clean state.

**Pull request closed without merging**